tests/suites/test_suite_x509write.function

```diff
@@ -377,12 +377,19 @@ void x509_crt_check( char *subject_key_file, char *subject_pwd,
     if( pk_wrap == 2 )
         TEST_ASSERT( mbedtls_pk_get_type( &issuer_key ) == MBEDTLS_PK_OPAQUE );
 
+#if !defined(MBEDTLS_BIGNUM_C)
```

Comment:
This is test code, so here the situation is different from the example program: we want to test both the new function and the deprecated one (but that doesn't have to be in the same build).

I think the correct way of achieving this is to use `MBEDTLS_TEST_DEPRECATED`, in this case it should look like:
```c
#if defined(MBEDTLS_TEST_DEPRECATED) && defined(MBEDTLS_BIGNUM_C)
// use old function
#else
// use new function
#endif
```

Also, we should only declare the MPI (which I suggest renaming to `serial_mpi` if you also do the other renaming I'm suggesting) under the same conditions.